src/soft-wraps-transform.coffee

```diff
@@ -19,15 +22,33 @@ class SoftWrapsTransform
       if /\s/.test(@source.characterAt(sourcePosition))
         sourceEndPosition = nextSourcePosition
       else if width > @clientWidth
-        targetTraversal = Point(1, 0)
-        break
+        width = 0
+        sourceTraversal = sourceEndPosition.traversalFrom(start)
+        regions.push new Region(sourceTraversal, Point(1, 0))
+        start = sourceEndPosition
+        if @hasTabAt(sourceStartPosition)
```

Comment:
So continuing my thoughts from below... here I think you would precompute the number of leading whitespaces on the source line before you started soft-wrapping it, then just insert the appropriate region for each soft-wrap.

The reason I used the string `'exclusive'` before was to try to express the clipping rules. In the case of these leading whitespace regions, we'd need to also include the region's start point in the clipping because you shouldn't be allowed to place the cursor at the beginning of the soft-wrap indentation.